pyccel/ast/numpyext.py

```diff
@@ -786,22 +786,22 @@ def __new__(cls, a, dtype=None, order='K', subok=True, shape=None):
 class NumpyNorm(PyccelInternalFunction):
     """ Represents call to numpy.norm"""
     _dtype = NativeReal()
+    _shape = ()
+    _rank  = 0
 
     def __init__(self, arg, dim=None):
-        if isinstance(dim, ValuedArgument):
-            dim = dim.value
-        if self.dim is not None:
-            sh = list(sh)
-            del sh[self.dim]
-            self._shape = tuple(sh)
-        else:
-            self._shape = ()
-        self._rank = len(self._shape)
         super().__init__(arg, dim)
+        if isinstance(arg.dtype, NativeBool):
+            arg = PythonFloat(PythonInt(arg))
+        elif isinstance(arg.dtype, NativeComplex):
+            arg = PythonAbs(arg)
+        elif not isinstance(arg.dtype, NativeReal):
+            arg = PythonFloat(arg)
```

Comment:
Presumably this is done to change the precision? Why is this necessary? Does `np.norm` not match the precision of its results to the precision of its arguments?

Reply:
Never mind, there is a `not`